modules/core/include/opencv2/core/cvdef.h

```diff
@@ -298,6 +298,30 @@ enum CpuFeatures {
 #  define CV_VFP 0
 #endif
 
+#ifndef CV_THREAD_LOCAL
```

Comment:
This is not needed in public headers.

Also we use [cv::TLSData wrapper](https://github.com/opencv/opencv/blob/3.2.0/modules/core/include/opencv2/core/utility.hpp#L634) over of "thread_local".